joinfaces-autoconfigure/src/main/java/org/joinfaces/autoconfigure/adminfaces/AdminfacesAutoConfiguration.java

```diff
@@ -0,0 +1,119 @@
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Scope;
+import org.springframework.context.annotation.ScopedProxyMode;
+import org.springframework.http.HttpStatus;
+import org.springframework.lang.Nullable;
+
+/**
+ * Spring Boot Auto Configuration of AdminFaces.
+ *
+ * @author Marcelo Fernandes
+ */
+@Configuration
+// will adminfaces autoconfigure via application.yml ?
+//@EnableConfigurationProperties(AdminfacesProperties.class)
+@ComponentScan({"com.github.adminfaces.template.bean",
+	"com.github.adminfaces.template.config",
+	"com.github.adminfaces.template.security"})
+@ServletComponentScan({"com.github.adminfaces.template.security",
+	"com.github.adminfaces.template.session"})
+@ConditionalOnClass(AdminSession.class)
+@AutoConfigureBefore(PrimefacesAutoConfiguration.class)
+@ConditionalOnWebApplication(type = ConditionalOnWebApplication.Type.SERVLET)
+public class AdminfacesAutoConfiguration {
+
+	@Bean
+	public BeanPostProcessor adminfacesPrimeFacesPropertiesPostProcessor() {
+		return new PrimeFacesPropertiesPostProcessor();
+	}
+
+	// AdminSession does not contain @Named.
+	@Bean
+	@Scope(value = "session", proxyMode = ScopedProxyMode.TARGET_CLASS)
+	public AdminSession adminSession() {
+		return new AdminSession();
+	}
+
+	/**
+	 * This {@link WebServerFactoryCustomizer} adds a {@link ServletContextInitializer} to the embedded servlet-container
+	 * which is equivalent to adminfaces's own {@code META-INF/web-fragment.xml}.
+	 * @return adminfaces web server factory customizer
+	 */
+	@Bean
+	public WebServerFactoryCustomizer<ConfigurableServletWebServerFactory> adminfacesWebServerFactoryCustomizer() {
+		return factory -> {
+			factory.addErrorPages(new ErrorPage(HttpStatus.FORBIDDEN, "/403.jsf"),
+					new ErrorPage(AccessDeniedException.class, "/403.jsf"),
+					new ErrorPage(HttpStatus.NOT_FOUND, "/404.jsf"),
+					new ErrorPage(HttpStatus.INTERNAL_SERVER_ERROR, "/500.jsf"),
+					new ErrorPage(Throwable.class, "/500.jsf"),
+					new ErrorPage(ViewExpiredException.class, "/expired.jsf"),
+					new ErrorPage(OptimisticLockException.class, "/optimistic.jsf")
+			);
+			factory.addInitializers(servletContext -> {
+				servletContext.addListener(new AdminServletContextListener());
+			});
+		};
+	}
+
+	/**
+	 * Configures Primefaces to use admin theme.
+	 *
+	 * @author Marcelo Fernandes
+	 */
+	static class PrimeFacesPropertiesPostProcessor implements BeanPostProcessor {
+		@Override
+		public Object postProcessBeforeInitialization(@Nullable Object bean, @Nullable String beanName) throws BeansException {
+			if (bean instanceof Primefaces4_0Properties) {
+				Primefaces4_0Properties properties = (Primefaces4_0Properties) bean;
+				properties.setTheme("admin");
+			}
+			if (bean instanceof Primefaces5_2Properties) {
+				Primefaces5_2Properties properties = (Primefaces5_2Properties) bean;
+				properties.setFontAwesome(true);
```

Comment:
I think we should check if the user has already provided a value and log an info if we change it.

Reply:
ok